### ISSUE TITLE
Assistance Needed: Restricting "Advanced Company Settings" in Version 4.2

### ISSUE THREAD

**Issue by eamon55** [iomad/iomad#2046]
Dear support team,

I am currently using version 4.2 and have encountered an issue regarding the Advanced Company Settings. In the setup, options such as Maximum active users and Contract end date, when configured by the Company Manager, can also be modified by the Department Training Manager.

Unfortunately, under Restrict Capabilities, there doesn’t appear to be an option to restrict access to Advanced Company Settings specifically. This creates a challenge as we would prefer to limit these settings to the Company Manager role only.

Could you please advise if there is a solution or workaround to address this? Alternatively, are there any updates or planned features that could resolve this limitation?

Thank you for your support, and I look forward to your guidance.

**Comment by turf212** [comment]
Company managers probably shouldn't be editing those fields either.  It's more for controlling the companies within the site and if they have access then they can get around the controls which are meant to be applied to them.

In any case - there is a new capability added block/iomad_company_admin:company_edit_restricted and I've put maxuser, duration. suspend after and hostname behind it in the form.

**Comment by eamon55** [comment]
Subject: Error Encountered After Update  

Hi Derick,  

Thank you for the update.  

After the recent update, we encountered the following error when attempting to install IOMAD 4.1 or 4.2:  

```
Exception - iomad::has_capability(): Argument #2 ($context) must be of type context, null given, called in [dirroot]/blocks/iomad_company_admin/classes/forms/company_edit_form.php on line 206  
```

For reference, here’s the link with more details: [[More information about this error](http://docs.moodle.org/401/en/error/moodle/generalexceptionmessage)](http://docs.moodle.org/401/en/error/moodle/generalexceptionmessage).  

Could you please advise on how to resolve this issue?  

Best regards,

**Comment by turf212** [comment]
Sorry - the initial fix was brought down from 4.3 - which uses the company context so the line shoudl have been $context and not $this->context.  I've pushed a fix for 4.1 and 2

**Comment by eamon55** [comment]
Thank you for the update. After the last update, the previous error has disappeared, which is great. However, when I click "Create User" and attempt to save, it is not working.

Could you please assist with resolving this issue?

Best regards,
